system/lib/dlmalloc.c

```diff
@@ -4386,7 +4401,7 @@ static int sys_trim(mstate m, size_t pad) {
                         }
                     }
                 }
-                else if (HAVE_MORECORE) {
+                else if (HAVE_MORECORE && !UNSIGNED_MORECORE) {
```

Comment:
What does this part of the change do?

Reply:
This prevents dlmalloc from trimming memory (calling morecore with a negative value) when UNSIGNED_MORECORE.

Reply:
If so, perhaps this should check `MORECORE_CANNOT_TRIM`?

Reply:
You are totally right.

Reply:
@sbc100, what do you prefer?:
* Using `#ifndef MORECORE_CANNOT_TRIM`
* Unconditionally define `MORECORE_CANNOT_TRIM` and then using it inline just like `UNSIGNED_MORECORE`

I ask because I originally changed it so that it was always defined but you correctly suggested to undo that because it wasn't really needed at that time.

Reply:
Seems like using `#ifndef MORECORE_CANNOT_TRIM` is more inline with the original code?